specs/select.spec.md

```diff
@@ -0,0 +1,158 @@
+# Select + Select-Option
+
+## Overview
+
+The `select` is a component that provides a list of options for the user to select from.
+
+### Background
+
+There is a standard select html control, `<select>` and `<option>`, but it doesn't support the full level of detail, styling, etc that most design systems need or require. For context please see [MDN Docs](https://developer.mozilla.org/en-US/docs/web/html/element/select).
+
+### Use Cases
+
+- *A customer using the component on a web page.*
+On a web page a customer is shopping for a new shirt in size medium. The customer opens the size `select` and sees a list of `<option>`s and selects size medium.
+
+### Features
+- form association
+- focus delegation
+
+### Features
+- Select one or multiple `<fast-option/>`(s) using the `multiple` attribute
+- When Dropdown does not have enough screen real estate to open below `<fast-select>` it will open above.
+
+### Prior Art/Examples
+- [FAST Select (React)](https://www.npmjs.com/package/@microsoft/fast-components-react-msft)
+- [Ant Design](https://ant.design/components/select/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/select/code/)
+- [Atlassian UI](https://atlaskit.atlassian.com/packages/core/select)
+- [Lightning Design System](https://www.lightningdesignsystem.com/components/select/)
+- [W3 Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-select`
+
+*Attributes:*
+- `autocomplete` - Allows the developer to provide a hint on how to search the content within the `<fast-option>`(s)
+- `autofocus` - Automatically focuses the control
+- `disabled` - Disables the control
+- `length` - Number of `<fast-option>` elements in the `<fast-select>` element
+- `menu-open` -If the menu of the `<fast-option>` is open or not
+- `multiple` - Allows user to select more than one `<fast-option>`
+- `name` - Name of the control
+- `options` - Returns an array of the `<fast-option>` elements contained by the `<fast-select>` element
```

Comment:
Are we expecting this list of options to be updated at any point? If you are the use of index for `selected-index` and `selected-options` may be problematic.

Reply:
We use this type of language elsewhere like `active-id` or `selected-index` the are attributes that can be updated in the DOM when a selection change is made. It can be initially set by the app author. Tabs is a goos example of this:

<img width="802" alt="Screen Shot 2020-08-12 at 9 02 00 AM" src="https://user-images.githubusercontent.com/37851214/90038724-9b6db300-dc7a-11ea-9637-dc621ae0a6f6.png">
<img width="711" alt="Screen Shot 2020-08-12 at 9 01 55 AM" src="https://user-images.githubusercontent.com/37851214/90038734-9d377680-dc7a-11ea-9686-ef4fad631320.png">

Reply:
Sure, the difference is I would expect tabs to come with the tab and tabpanel already included. I might expect a call to a server to populate a list of select options.

Reply:
We probably should just stick with `slectedOptions` then. @gregwhitworth, thoughts here?